File: fp_py/src/main/fruitpunch/native/common/number_event.py
import libfp_core
from fruitpunch.native.native_class import NativeClass
from fruitpunch.native.common.event import Event


class NumberEvent(Event):
    
    def __init__(self, name="undefined"):
        NativeClass.__init__(self)
        
        self.store(libfp_core.NumberEvent(name))
        
    def get_number(self):
        return self.run_native("get_number")
    
    def set_number(self, number):
        return self.run_native("set_number", number)
    
    number = property(get_number, set_number)
    
NativeClass.class_map[libfp_core.NumberEvent] = NumberEvent

File: fp_core/src/main/src/Common/NumberEvent.cpp
/*
 * NumberEvent.cpp
 *
 *  Created on: Aug 20, 2013
 *      Author: leo
 */

#include <fruitpunch/Common/NumberEvent.h>
#include <fruitpunch/Common/Events.h>

namespace fp_core {
namespace common {

NumberEvent::NumberEvent(std::string eventName) :
		Event(eventName),
		mNumber(-1) {

}

NumberEvent::~NumberEvent() {

}

int NumberEvent::getNumber() const {
	return mNumber;
}

void NumberEvent::setNumber(int number) {
	mNumber = number;
}

}
}


